internal/ssm_test.go

```diff
@@ -4,23 +4,28 @@ import (
 	"testing"
 	"time"
 
+	"gotest.tools/assert"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/ssm"
 	"github.com/aws/aws-sdk-go/service/ssm/ssmiface"
 )
 
-type GetSSMParameterTestValue struct {
-	path          string
-	required      bool
-	decrypt       bool
-	expectedValue string
+var fakeValue = "value"
+var fakeOtherValue = "other-value"
+var fakeMissingValue = "missing-value"
+
+type SSMParameter struct {
+	value         *string
+	defaultValue  *string
+	expectedValue *string
 }
 
-var ssmTestValues = map[string]GetSSMParameterTestValue{
-	"/root/parameter1": GetSSMParameterTestValue{"/root/parameter1", false, false, "value1"},
-	"/root/parameter2": GetSSMParameterTestValue{"/root/parameter2", false, false, "value2"},
-	"/root/parameter3": GetSSMParameterTestValue{"/root/parameter3", false, false, "value3"},
-	"/root/parameter4": GetSSMParameterTestValue{"/root/parameter4", false, false, "value4"},
-	"/root/parameter5": GetSSMParameterTestValue{"/root/parameter5", false, false, "value5"},
+var fakeSSMStore = map[string]SSMParameter{
```

Comment:
these are just 4 cases, I would have 4 separate tests(eventually using a fixture) rather then a map of input -> output

Reply:
What do you mean? I refactored this 5 times already.
* Go does not have generics so you would have to hack stuff around like having multiple mocks
* I do not see a disadvantage of using this approach, we are doing basic in out expectancy tests.